### ISSUE TITLE
UI Tests check is flaky on CI

### ISSUE THREAD

**Issue by jtpio** [jupyterlab/jupyterlab#14550]
## Description

UI Tests check have been flaky on CI recently.

## Reproduce

For example: https://github.com/jupyterlab/jupyterlab/actions/runs/4984379184

Does not show any failure in the report: 

![image](https://github.com/jupyterlab/jupyterlab/assets/591645/0ba66eab-77ae-44b8-bf92-58e8baeff8ed)

But in the logs:

![image](https://github.com/jupyterlab/jupyterlab/assets/591645/1aacf36c-e477-4a91-9546-c52204934e34)

## Expected behavior

It should pass.

## Context

This adds noise when reviewing  PRs and would be great to fix.

**Comment by krassowski** [comment]
I think I had already fixed on of those with changes in https://github.com/jupyterlab/jupyterlab/pull/14421 but I will try to separate the changes out.

**Comment by krassowski** [comment]
On the other hand I see this test to fail frequently:

`test/benchmark/notebook.spec.ts:94:9 › Benchmark › measure large_code_notebook.ipynb`

![image](https://github.com/jupyterlab/jupyterlab/assets/5832902/56ff6fbe-d1de-4672-8398-6a365fcc6e79)

**Comment by krassowski** [comment]
Of note, a few of these failures seem to be related to the transition for "defer" windowing mode in https://github.com/jupyterlab/jupyterlab/pull/14490

#### Notebook Search › Search within outputs

Here a dialog "notebook is not fully rendered yet, do you want to continue" shows up which was not the case previously. Adding a conditional check to accept the dialog fixed falkiness in https://github.com/jupyterlab/jupyterlab/pull/14421

![image](https://github.com/jupyterlab/jupyterlab/assets/5832902/886d6aa8-51cf-4ec9-ac1d-34a7906758cf)

#### Benchmark › measure large_code_notebook.ipynb

Here the notebook is clearly not fully rendered yet:

| Expected | Actual |
|--|--|
| ![large-code-notebook-ipynb-expected](https://github.com/jupyterlab/jupyterlab/assets/5832902/08f12423-c476-4ac6-9fe6-3ba8dbac68fb) | ![large-code-notebook-ipynb-actual](https://github.com/jupyterlab/jupyterlab/assets/5832902/e827ec7d-c4de-43fc-a455-b4fcaceadfd0) |

### Notebook Search › Search on typing

This one may be an interaction with "defered" mode too, but more likely just needs an await to confirm that search has executed before we start editing cell here:

https://github.com/jupyterlab/jupyterlab/blob/3fd8c726a22c636a447133345b1c1c27fc8a5674/galata/test/jupyterlab/notebook-search.test.ts#L457

| Results in active yellow highlight | Results in no active highlight |
|--|--|
| [path-1.webm](https://github.com/jupyterlab/jupyterlab/assets/5832902/854dc05d-5af8-4ec4-a96e-162a122d3461) | [path-2.webm](https://github.com/jupyterlab/jupyterlab/assets/5832902/5e82ece2-fd36-4525-884e-bc57ab821e03) |

**Comment by krassowski** [comment]
The benchmark failures seem to stem from the spinner being hidden before the deferred cells are shown. This is expected, but then I am not sure how we want to update the test to account for that. The relevant lines are:

https://github.com/jupyterlab/jupyterlab/blob/3fd8c726a22c636a447133345b1c1c27fc8a5674/galata/test/benchmark/notebook.spec.ts#L111-L131
